api/src/service/domain/workflow/workflowitem_update.ts

```diff
@@ -89,7 +113,46 @@ export async function updateWorkflowitem(
       ),
     );
 
-  return { newEvents: [newEvent, ...notifications], workflowitem: result };
+  // handle new documents
+  try {
+    let newDocumentUploadedEvents: BusinessEvent[] = [];
+    if (newEvent.update.documents && newEvent.update.documents.length > 0) {
+      const { documents } = newEvent.update;
+
+      newDocumentUploadedEvents = documents.map((d, i) => {
+        const docToUpload: UploadedDocument = {
+          base64: modification.documents ? modification.documents[i].base64 : "",
```

Comment:
why is the value for `base64` set to empty string if there is no document at all? should this not be an error?

Reply:
the `newEvent` or the prop `documents` are built from the modification.documents but they may not include the base64 so the base64 value has to be get from the modification. There is no way  the documents in the modification can be empty. So the empty string acts like a placeholder. The check has to be made since typescript compiler warns that modification.documents could potentially be empty.